services/tasks/LocalJob.go

```diff
@@ -375,12 +375,13 @@ func (t *LocalJob) getPlaybookArgs(username string, incomingVersion *string) (ar
 		}
 	}
 
-	if t.Task.Limit != "" {
-		t.Log("--limit=" + t.Task.Limit)
-		taskExtraArgs = append(taskExtraArgs, "--limit="+t.Task.Limit)
+	if t.Task.Limit != nil && len(*t.Task.Limit) > 0 {
+		t.Log("--limit=" + *t.Task.Limit)
+		taskExtraArgs = append(taskExtraArgs, "--limit="+*t.Task.Limit)
 	}
 
-	args = append(args, templateExtraArgs...)
+	// override the args from task to template
+	// args = append(args, templateExtraArgs...)
```

Comment:
I implemted a new function called IsCLIArgOverriden that compares the task and template cli args one by one to each other and manages to decide if an template argument is being overriden by a task cli argument. The what it does it, replacing the CLI argument of `templateExtraArgs` with task argument that is targeting to override it. This way task arguments are redundent after the comparison because we are managing to override them to templateExtraArgs variable and use that only. This method should be covering any way of running a template.